cmd/snap/complete.go

```diff
@@ -0,0 +1,181 @@
+package main
+
+import (
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/jessevdk/go-flags"
+
+	"github.com/snapcore/snapd/asserts"
+	"github.com/snapcore/snapd/client"
+)
+
+type installedSnapName string
+
+func (s installedSnapName) Complete(match string) []flags.Completion {
+	cli := Client()
+	snaps, err := cli.List(nil, nil)
+	if err != nil {
+		return nil
+	}
+
+	ret := make([]flags.Completion, 0, len(snaps))
+	for _, snap := range snaps {
+		if strings.HasPrefix(snap.Name, match) {
```

Comment:
It would be nice to have a helper function that just does this internally but could be improved to say, do fuzzy searching or smart suggestions given some context.